vision/use-cases/image_segmentation_layers.ipynb

```diff
@@ -0,0 +1,961 @@
+      "metadata": {
+        "id": "FPMmGO5TFLYG"
+      },
+      "outputs": [],
+      "source": [
+        "from IPython.display import display\n",
+        "\n",
+        "foreground_checkbox = widgets.Checkbox(\n",
+        "    value=True, description=\"Foreground Mask\", disabled=False\n",
+        ")\n",
+        "background_checkbox = widgets.Checkbox(\n",
+        "    value=True, description=\"Background Mask\", disabled=False\n",
+        ")\n",
+        "semantic_checkbox = widgets.Checkbox(\n",
+        "    value=True, description=\"Semantic Mask\", disabled=False\n",
+        ")\n",
+        "prompt_checkbox = widgets.Checkbox(\n",
+        "    value=True, description=\"Prompt Mask\", disabled=False\n",
+        ")\n",
+        "\n",
+        "\n",
+        "display(foreground_checkbox)\n",
+        "display(background_checkbox)\n",
+        "display(semantic_checkbox)\n",
+        "display(prompt_checkbox)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "rFAh_A01gc_G"
+      },
+      "source": [
+        "### Add selected mask images as layers\n",
+        "\n",
+        "Once the layers are added, you will save the final PSD file."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 13,
+      "metadata": {
+        "id": "VXCM_QOz3Uln"
+      },
+      "outputs": [],
+      "source": [
+        "with Wand_Image(filename=\"original.png\") as img:\n",
+        "    img.read(filename=\"original.png\")\n",
+        "\n",
+        "    if foreground_checkbox.value:\n",
+        "        img.read(filename=\"foreground.png\")\n",
+        "    if background_checkbox.value:\n",
+        "        img.read(filename=\"background.png\")\n",
+        "    if semantic_checkbox.value:\n",
+        "        img.read(filename=\"semantic.png\")\n",
+        "    if prompt_checkbox.value:\n",
+        "        pattern = re.compile(\"prompt*\")\n",
+        "        for file in os.listdir(\".\"):\n",
+        "            if pattern.match(file):\n",
+        "                img.read(filename=file)\n",
+        "\n",
+        "    img.save(filename=\"output.psd\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C--fF0Wjg8Dp"
+      },
+      "source": [
+        "### Upload the PSD file to Google Cloud Storage bucket"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "LW4VpNDsemLm"
+      },
+      "outputs": [],
+      "source": [
+        "prefix = \"psd_\" + str(randrange(10000, 100000))\n",
+        "bucket_name = \"[your-bucket-name]\"  # @param {type:\"string\"}\n",
+        "\n",
+        "storage_client = storage.Client()\n",
+        "bucket = storage_client.bucket(bucket_name)\n",
+        "blob = bucket.blob(prefix)\n",
+        "\n",
+        "blob.upload_from_filename(\"output.psd\")\n",
+        "print(\"Uploaded \" + prefix)"
```

Comment:
Does the PSD file really need to be uploaded to GCS?

Reply:
This is a requested feature for the use case.